server/src/main/java/org/opensearch/index/mapper/DocumentParser.java

```diff
@@ -454,6 +456,7 @@ private static void innerParseObject(
             }
             token = parser.nextToken();
         }
+        context.decrementFieldCurrentDepth();
```

Comment:
Do we need to decrement in case of an exception?

Reply:
> Add to CHANGELOG. We'll also need to document this.
> 
> How does this alter the default behavior? If I understand correctly now the mapping depth limit is now enforced - what is its default value?

Default behaviour remains same as long as depth of a field object is under "index.mapping.depth.limit". Default value for the limit is 20. If an object depth in the request body crosses this limit then we throw parse_exception.

Reply:
> Do we need to decrement in case of an exception?
The context object is discarded when an exception occurs while parsing the document. This is done in ["DocumentParser::parseDocument"](https://github.com/opensearch-project/OpenSearch/blob/d266a732aea4cf87781acf93e92c56ad3f1d0913/server/src/main/java/org/opensearch/index/mapper/DocumentParser.java#L89). So, decrementing in case of exception isn't necessary.